### PR TITLE
Added sort for tube stock plate along with dodgy test.

### LINKED ISSUE #640: GPL-867 BUG - 'input plate barcode' inconsistent for Eagle samples
From RT #712371:

> Today lab staff noticed that when processing plate through limber, it changed input plate barcode from EGC000040 (DN771253N) to EGC000080 (which is one of the quadrants). This is really confusing for the lab people.

> Here’s the last page in Limber when correct input plate is visible (LTHR-384 Lib PCR pool) – EGC000040: http://limber.psd.sanger.ac.uk/limber_plates/9fae9930-6c66-11eb-b5ab-fa163eac3af7

> When clicking on children, it leads us to page of LTHR-384 Pool XP: http://limber.psd.sanger.ac.uk/limber_tubes/f58eb95a-6c68-11eb-b5ab-fa163eac3af7 where suddenly input plate barcode changes to EGC000080.

> I checked few other released Eagle plates and the same happen for each of them in LTHR-384 Pool XP step.

> Is that intended behaviour? If not, can it be fixed for EGC/EGL plates?

From James (Slack):

> The fix I applied to input plates recently will also need to be applied to the same method on tubes. (ie. the order by Id one)

(He's talking about https://github.com/sanger/limber/issues/619)

Also raised separately in RT #712672

### PR DESCRIPTION
**Pull request by stevieing**
Closes #640 

Changes proposed in this pull request:

*
*
* ...

### CODE REVIEW

File: app/sequencescape/sequencescape/api/v2/tube.rb [hunk 1]
```diff
@@ -52,7 +52,9 @@ def barcode
   end
 
   def stock_plate(purpose_names: SearchHelper.stock_plate_names)
-    @stock_plate ||= ancestors.where(purpose_name: purpose_names).last
+    # this is an array not a collection so cant use order_by
+    # sort_by naturally sorts in ascending order
```
Comment (KatyTaylor) [inline app/sequencescape/sequencescape/api/v2/tube.rb:56]:
```suggestion
    # max_by naturally sorts in ascending order
```

File: spec/sequencescape/api/v2/tube_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    let(:tube_with_ancestors)   { create :v2_tube, barcode_number: 12_345, ancestors: stock_plates }
+
+    it 'should return the last plate' do
+      expect(stock_plates.length).to eq(4)
+    end
+
+    # I know this is a real hack but all we need to know is whether
+    # it returns the last stock plate
+    # I am not going to fumble about trying to recreate the whole pipeline
+    it 'should have some ancestors' do
```
Comment (KatyTaylor) [inline spec/sequencescape/api/v2/tube_spec.rb:32]: Feel like you've got your test names swapped around? This one should say 'should return the last plate' and the other one 'should have some ancestors' ?

**Review by KatyTaylor — APPROVED** [review]
Approved, just a couple of 'naming' things.

**Comment by KatyTaylor** [comment]
Did you see this commit that James did after the fact? https://github.com/sanger/limber/pull/623/files

Wondering if you need this too

**Comment by stevieing** [comment]
> Did you see this commit that James did after the fact? https://github.com/sanger/limber/pull/623/files
> 
> Wondering if you need this too

I did see that but not sure what you mean? If you mean returning the stock plate then I was under the impression that it is a tube so that wouldn't work. If you mean the tests then I did start trying to use the stuff that James had and I was going down a rabbit hole so found the easy way out!

File: spec/sequencescape/api/v2/tube_spec.rb (hunk 2 above)
Comment (stevieing) [reply spec/sequencescape/api/v2/tube_spec.rb:32]: It only needs one test so will remove the other one.

File: app/sequencescape/sequencescape/api/v2/tube.rb (hunk 1 above)
Comment (stevieing) [reply app/sequencescape/sequencescape/api/v2/tube.rb:56]: Rubocop did that!

**Pull request merged**